azure-storage-queue/src/main/scala/akka/stream/alpakka/azure/storagequeue/AzureQueueSourceStage.scala

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (C) 2016-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.stream.alpakka.azure.storagequeue
+
+import com.microsoft.azure.storage.queue.{CloudQueue, CloudQueueMessage}
+import akka.stream.stage.{GraphStage, GraphStageLogic, OutHandler, TimerGraphStageLogic}
+import akka.stream.{Attributes, Outlet, SourceShape}
+import akka.actor.ActorSystem
+import scala.collection.mutable.Queue
+import scala.concurrent.duration._
+import akka.NotUsed
+import akka.stream.impl.Stages.DefaultAttributes.IODispatcher
+
+/** Settings for AzureQueueSource
+ *
+ * @param initalVisibilityTimeout Specifies how many seconds a message becomes invisible after it has been dequeued.
+ *        See parameter of the same name in [[com.microsoft.azure.storage.queue.CloudQueue$.retrieveMessages]].
+ * @param batchSize Specifies how many message are fetched in one batch.
+ *        (This is the numberOfMessages parameter in [[com.microsoft.azure.storage.queue.CloudQueue$.retrieveMessages]].)
+ * @param retrieveRetryTimeout If None the [[AzureQueueSource]] will be completed if the queue is empty.
+ *        If Some(timeout) [[AzureQueueSource]] will retry after timeout to get new messages. Do not set timeout to low.
+ */
+final case class AzureQueueSourceSettings(
+    initialVisibilityTimeout: Int,
+    batchSize: Int,
+    retrieveRetryTimeout: Option[FiniteDuration]
+)
+
+object AzureQueueSourceSettings {
+  // default initialVisibilityTimeout (30) is from
+  // com.microsoft.azure.storage.queue.QueueConstants.DEFAULT_VISIBILITY_MESSAGE_TIMEOUT_IN_SECONDS
+  val default = AzureQueueSourceSettings(30, 10, None)
```

Comment:
Good comment, make it a Scaladoc instead! Also, in Akka constants in Scala use uppercase as first letter, so `Default`